modules/ktx/texture_loader_ktx.cpp

```diff
@@ -0,0 +1,561 @@
+					format = Image::FORMAT_ETC2_R11S;
+					break;
+				case VK_FORMAT_EAC_R11G11_UNORM_BLOCK:
+					format = Image::FORMAT_ETC2_RG11;
+					break;
+				case VK_FORMAT_EAC_R11G11_SNORM_BLOCK:
+					format = Image::FORMAT_ETC2_RG11S;
+					break;
+				case VK_FORMAT_ETC2_R8G8B8_UNORM_BLOCK:
+					format = Image::FORMAT_ETC2_RGB8;
+					break;
+				case VK_FORMAT_ETC2_R8G8B8A8_UNORM_BLOCK:
+					format = Image::FORMAT_ETC2_RGBA8;
+					break;
+				case VK_FORMAT_ETC2_R8G8B8A1_UNORM_BLOCK:
+					format = Image::FORMAT_ETC2_RGB8A1;
+					break;
+#endif
+				case VK_FORMAT_ASTC_4x4_SRGB_BLOCK:
+					format = Image::FORMAT_ASTC_4x4;
+					break;
+				case VK_FORMAT_ASTC_4x4_UNORM_BLOCK:
+					format = Image::FORMAT_ASTC_4x4_HDR;
+					break;
+				case VK_FORMAT_ASTC_8x8_SRGB_BLOCK:
+					format = Image::FORMAT_ASTC_8x8;
+					break;
+				case VK_FORMAT_ASTC_8x8_UNORM_BLOCK:
+					format = Image::FORMAT_ASTC_8x8_HDR;
+					break;
+				default:
+					ktxTexture_Destroy(ktx_texture);
+					ERR_FAIL_V_MSG(Ref<Resource>(), "Unsupported format " + itos(((ktxTexture2 *)ktx_texture)->vkFormat) + " of KTX2 texture file '" + p_path + "'.");
+					break;
+			}
+			break;
+		}
+		default:
+			ktxTexture_Destroy(ktx_texture);
+			ERR_FAIL_V_MSG(Ref<Resource>(), "Unsupported version KTX texture file '" + p_path + "'.");
+			break;
+	}
+
+	Vector<uint8_t> src_data;
+
+	/* KTX use 4-bytes padding, don't use mipmaps if padding is effective
+	 * XXX unpad dynamically
+	 */
+	int pixel_size = Image::get_format_pixel_size(format);
+	int pixel_rshift = Image::get_format_pixel_rshift(format);
+	int block = Image::get_format_block_size(format);
+	int minw, minh;
+	Image::get_format_min_pixel_size(format, minw, minh);
+	int w = width;
+	int h = height;
+	for (uint32_t i = 0; i < mipmaps; ++i) {
+		ktx_size_t mip_size = ktxTexture_GetImageSize(ktx_texture, i);
+		size_t bw = w % block != 0 ? w + (block - w % block) : w;
+		size_t bh = h % block != 0 ? h + (block - h % block) : h;
+		size_t s = bw * bh;
+		s *= pixel_size;
+		s >>= pixel_rshift;
+		if (mip_size != static_cast<ktx_size_t>(s)) {
+			if (!i) {
+				ktxTexture_Destroy(ktx_texture);
+				ERR_FAIL_V_MSG(Ref<Resource>(), "Unsupported padded KTX texture file '" + p_path + "'.");
+			}
+			mipmaps = 1;
+			break;
+		}
+		w = MAX(minw, w >> 1);
+		h = MAX(minh, h >> 1);
+	}
+
+	for (uint32_t i = 0; i < mipmaps; ++i) {
+		ktx_size_t mip_size = ktxTexture_GetImageSize(ktx_texture, i);
+		ktx_size_t offset;
+		if (ktxTexture_GetImageOffset(ktx_texture, i, 0, 0, &offset) != KTX_SUCCESS) {
+			ktxTexture_Destroy(ktx_texture);
+			ERR_FAIL_V_MSG(Ref<Resource>(), "Invalid KTX texture file '" + p_path + "'.");
+		}
+		int prev_size = src_data.size();
+		src_data.resize(prev_size + mip_size);
+		memcpy(src_data.ptrw() + prev_size, ktxTexture_GetData(ktx_texture) + offset, mip_size);
+	}
+
+	Ref<Image> img = memnew(Image(width, height, mipmaps - 1, format, src_data));
+	if (srgb)
+		img->srgb_to_linear();
```

Comment:
Please make sure to always use braces.
```suggestion
	if (srgb) {
		img->srgb_to_linear();
	}
```